### ISSUE TITLE
NullPointerException when accessing node details

### ISSUE THREAD

**Issue by alexanderteves** [Graylog2/graylog2-web-interface#1517]
Running both server and web v1.1.4 in Ubuntu (ElasticSearch v1.6.0), I get the following NullPointerException when I click the only node's details on any part of the web interface. Everything else works flawlessly, even heap usage is displayed correctly and the metrics page is accessible.

```
2015-07-07T13:42:53.747Z - [ERROR] - from play in play-akka.actor.default-dispatcher-3 
Cannot invoke the action, eventually got an error: java.lang.NullPointerException

2015-07-07T13:42:53.751Z - [ERROR] - from application in play-akka.actor.default-dispatcher-3 


! @6mlai55kh - Internal server error, for (GET) [/system/nodes/ebb19cce-4162-4e49-822f-3ed6c3591d5e] ->

play.api.Application$$anon$1: Execution exception[[NullPointerException: null]]
	at play.api.Application$class.handleError(Application.scala:296) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.DefaultApplication.handleError(Application.scala:402) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Promise$class.complete(Promise.scala:55) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.processBatch$1(BatchingExecutor.scala:67) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:82) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72) [org.scala-lang.scala-library-2.10.4.jar:na]
	at akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [org.scala-lang.scala-library-2.10.4.jar:na]
Caused by: java.lang.NullPointerException: null
	at views.html.system.nodes.show$.apply(show.template.scala:175) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at views.html.system.nodes.show$.render(show.template.scala:306) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at views.html.system.nodes.show.render(show.template.scala) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at controllers.NodesController.node(NodesController.java:106) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at Routes$$anonfun$routes$1$$anonfun$applyOrElse$43$$anonfun$apply$567.apply(routes_routing.scala:1947) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:na]
	at Routes$$anonfun$routes$1$$anonfun$applyOrElse$43$$anonfun$apply$567.apply(routes_routing.scala:1947) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:na]
	at play.core.Router$HandlerInvokerFactory$$anon$4.resultCall(Router.scala:264) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.invocation(Router.scala:255) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:55) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.GlobalSettings$1.call(GlobalSettings.java:67) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.mvc.Security$AuthenticatedAction.call(Security.java:44) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:40) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:32) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$.apply(Future.scala:31) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$.apply(Future.scala:485) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.j.JavaAction$class.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.apply(Router.scala:252) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:129) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:128) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:128) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:121) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	... 6 common frames omitted
```

**Comment by joschi** [comment]
Could be related to https://groups.google.com/forum/#!topic/graylog2/yDfh3JmiXvk

**Comment by alexanderteves** [comment]
Thank you. I have checked that post before while researching, it relates to the node overview. I investigated the state of and connection to MongoDB, which seems to be fine.

**Comment by joschi** [comment]
Please check and post the actual response from the Graylog node for the resource `/system/cluster/nodes/$NODE_ID` (replace `$NODE_ID` with the actual ID of the node you want to access).

You can get this by running the following `curl` command (replacing `$ADMIN_PASSWORD` and `$NODE_ID` with the respective information):
```
curl -u admin:$ADMIN_PASSWORD -X GET -i http://127.0.0.1:12900/system/cluster/nodes/$NODE_ID?pretty=true
```

**Comment by alexanderteves** [comment]
No problem. Seems to be what is expected:

```
HTTP/1.1 200 OK
Content-Type: application/json
X-Graylog-Node-ID: 3d2d1c52-3368-457d-aa94-2157c1971689
Vary: Accept-Encoding
X-Runtime-Microseconds: 823
Transfer-Encoding: chunked

{
    "node_id" : "3d2d1c52-3368-457d-aa94-2157c1971689",
    "type" : "server",
    "transport_address" : "http://127.0.0.1:12900/",
    "last_seen" : "2015-07-08T07:59:17.000Z",
    "short_node_id" : "3d2d1c52",
    "is_master" : true
}
```

**Comment by alexanderteves** [comment]
So I reproduced it with all 1.1.x versions of web and server and tried some other combinations (different Java version, OS, *.deb / *.tgz) but the result is the same. So I suspect a fundamental error in my configurations. I will just post them here, maybe there is something obvious.

server
```
is_master = true
node_id_file = /etc/graylog/server/node-id
password_secret = rBAEavvhQ50YGe[...]ClVspQg4nfBNV4EHBrEyT
root_username = admin
root_password_sha2 = 874c253ec99bd3d740d83[...]1df1cbb59dca50f22fb8a541a202656
rest_listen_uri = http://127.0.0.1:12900/
message_journal_enabled = false
mongodb_uri = mongodb://localhost/graylog
mongodb_useauth = false
elasticsearch_cluster_name = graylog
elasticsearch_max_docs_per_index = 7500000
elasticsearch_max_number_of_indices = 16
elasticsearch_shards = 1
elasticsearch_replicas = 0
elasticsearch_discovery_zen_ping_multicast_enabled = false
elasticsearch_discovery_zen_ping_unicast_hosts = 127.0.0.1:9300
```
web
```
graylog2-server.uris="http://127.0.0.1:12900/"
application.secret="k2mkkOZbk5kg[...]azRxgifz9NIW2XdP1xYO3JGA3iB1oKBKBn9jOmjfGAsP5P9M0Brc"
field_list_limit=100
application.global=lib.Global
```

As I said, everything else works out of the box, processes logs, ready for production. But thanks so far!

**Comment by joschi** [comment]
@alexanderteves `password_secret` in the Graylog configuration and `application.secret` in the web interface configuration need to be identical.

**Comment by alexanderteves** [comment]
@joschi I mixed up two installations when pasting. Passwords are identical, checked again. Sorry for the confusion.

**Comment by alexanderteves** [comment]
I extracted the server and web configurations from the official Graylog docker image. I compared them to my configurations:

Setting `message_journal_enabled` to `true` (and providing the `message_journal_dir`) made the node's details work. Reproducable by setting it to `false` again which results in the mentioned exception.

Besides, `password_secret` and `application.secret` differ there.

**Comment by joschi** [comment]
> Setting message_journal_enabled to true (and providing the message_journal_dir) made the node's details work. Reproducable by setting it to false again which results in the mentioned exception.

Interesting. Thanks for investigating and reporting this. We'll take a look at it!

**Comment by joschi** [comment]
Merged into `1.1` branch in 6c05cf3b209eff82427e5f180932b3a49d7ad908.
